patches/tModLoader/Terraria/ModLoader/ModBlockType.cs

```diff
@@ -170,4 +170,18 @@ public virtual void PlaceInWorld(int i, int j, Item item)
 	public virtual void ModifyLight(int i, int j, ref float r, ref float g, ref float b)
 	{
 	}
+
+	/// <summary>
+	/// Allows you to change what happens when this tile/wall is converted into another biome. Return false to stop the game from converting the tile normally. Returns true by default.
+	/// <para/> Purification powder uses a separate conversionType, as it doesn't convert hallowed tiles back to purity tiles. Be sure to check for <see cref="BiomeConversionID.PurificationPowder"/> as well as <see cref="BiomeConversionID.Purity"/> when handling corruption/crimson tiles.
+	/// <para/> You can use <see cref="WorldGen.ConvertTile(int, int, int, bool)"/> or <see cref="WorldGen.ConvertWall(int, int, ushort)"/> to automatically handle tile framing and multiplayer syncing.
+	/// </summary>
+	/// <param name="i"></param>
+	/// <param name="j"></param>
+	/// <param name="conversionType">The <see cref="ID.BiomeConversionID"/> of the conversion</param>
+	/// <returns></returns>
+	public virtual bool Convert(int i, int j, int conversionType)
```

Comment:
`from converting the tile normally`. Modded tiles don't normally have any conversions defined though?

I'm not really sure under what cases you would want to have a specific return value. Needs more thought as to whether one is needed at all.

Reply:
Yep, lets change it to a `void` return, and move it after the global hooks. That way the mod tile `Convert` implementation is the 'vanilla' implementation for this tile, and globals can cancel it just like they can for vanilla tiles.